projeuler22.rb

```diff
@@ -8,7 +8,7 @@
 class NameCalculator
   attr_reader :namelist
   def initialize(namelist)
-    @namelist = File.read(namelist).scan(/\w+/).sort
+    @namelist ||= File.read(namelist).scan(/\w+/).sort
```

Comment:
`@namelist` will  _always_ be `nil` here (since this is where you're actually setting the variable)
Therefore you don't need to have conditional assignment!